src/ModelIO/MDLAnimatedValueTypes.cs

```diff
@@ -0,0 +1,805 @@
+			}
+		}
+
+		public virtual void Reset (Quaterniond [] values, double [] times)
+		{
+			if (values == null)
+				throw new ArgumentNullException (nameof (values));
+			if (times == null)
+				throw new ArgumentNullException (nameof (times));
+
+			unsafe {
+				fixed (Quaterniond* valuesPtr = values)
+					MDLMemoryHelper.Reset<Quaterniond> ((IntPtr) valuesPtr, values.Length, times, _ResetWithDoubleQuaternionArray);
+			}
+		}
+
+		public virtual Quaternion [] GetQuaternionValues ()
+		{
+			var count = ElementCount * TimeSampleCount;
+			var timesArr = new Quaternion [(int) count];
+
+			unsafe {
+				fixed (Quaternion* arrptr = timesArr)
+					MDLMemoryHelper.FetchValues<Quaternion> ((IntPtr) arrptr, count, _GetFloatQuaternionArray);
+			}
+
+			return timesArr;
+		}
+
+		public virtual Quaterniond [] GetQuaterniondValues ()
+		{
+			var count = ElementCount * TimeSampleCount;
+			var timesArr = new Quaterniond [(int) count];
+
+			unsafe {
+				fixed (Quaterniond* arrptr = timesArr)
+					MDLMemoryHelper.FetchValues<Quaterniond> ((IntPtr) arrptr, count, _GetDoubleQuaternionArray);
+			}
+
+			return timesArr;
+		}
+	}
+
+	public partial class MDLAnimatedScalar {
+
+		public virtual void Reset (float [] values, double [] times)
+		{
+			if (values == null)
+				throw new ArgumentNullException (nameof (values));
+			if (times == null)
+				throw new ArgumentNullException (nameof (times));
+
+			unsafe {
+				fixed (float* valuesPtr = values)
+				fixed (double* timesPtr = times)
+					_ResetWithFloatArray ((IntPtr) valuesPtr, (IntPtr) timesPtr, (nuint) times.Length);
+			}
+		}
+
+		public virtual void Reset (double [] values, double [] times)
+		{
+			if (values == null)
+				throw new ArgumentNullException (nameof (values));
+			if (times == null)
+				throw new ArgumentNullException (nameof (times));
+
+			unsafe {
+				fixed (double* valuesPtr = values)
+				fixed (double* timesPtr = times)
+					_ResetWithDoubleArray ((IntPtr) valuesPtr, (IntPtr) timesPtr, (nuint) times.Length);
+			}
+		}
+
+		public virtual float [] GetFloatValues ()
+		{
+			var count = TimeSampleCount;
+			var timesArr = new float [(int) count];
+
+			unsafe {
+				fixed (float* ptr = timesArr)
+					_GetFloatArray ((IntPtr) ptr, count);
+			}
+			return timesArr;
+		}
+
+		public virtual double [] GetDoubleValues ()
+		{
+			var count = TimeSampleCount;
+			var timesArr = new double [(int) count];
+
+			unsafe {
+				fixed (double* ptr = timesArr)
+					_GetDoubleArray ((IntPtr) ptr, count);
+			}
+			return timesArr;
+		}
+	}
+
+	public partial class MDLAnimatedVector2 {
+
+		public virtual void Reset (Vector2 [] values, double [] times)
+		{
+			if (values == null)
+				throw new ArgumentNullException (nameof (values));
+			if (times == null)
+				throw new ArgumentNullException (nameof (times));
+			if (values.Length != times.Length)
+				throw new InvalidOperationException ($"The length of '{nameof (values)}' array and '{nameof (times)}' array must match.");
```

Comment:
Grammar: "The length of _the_ 'values' array and _the_ 'times' array must match."